### PR TITLE
Glossary V1

### PR DESCRIPTION
**Pull request by ArnaudDebray**
@narve can you double-check the definition of Fill-and-kill and Fill-or-kill ? It's what we have written down when you explained but I'm not sure it's perfectly consistent with the standard definition of FoK and Fak in the common trading language.

### CODE REVIEW

**Comment by narve** [comment]
We can add a link in the README so that the glossary is available from the html page.

**Comment by narve** [comment]
I've pushed a commit with some clarifications and corrections. 

I believe there are still a few terms that is not properly defined. I suggest that we remove these terms from this PR, and have a separate discussion on those, in order to not keep this PR running for too long. 

The terms we believe need further work are: 
Block order, curtailable block order, and Reservation Market. 

Agree, @cdmNSIDE / @ArnaudDebray ?

**Comment by ArnaudDebray** [comment]
Hello Narve,

ok to discuss and search for a better definition for these terms. I think "market price" might be clarified further as well. 

But I don't see the need to close this PR quickly. Is It a problem to keep it open until next week ?

**Comment by narve** [comment]
> Hello Narve,
> 
> ok to discuss and search for a better definition for these terms. I think "market price" might be clarified further as well.
> 
> But I don't see the need to close this PR quickly. Is It a problem to keep it open until next week ?

No, it is not a problem actually. This is a separate file so there will not be conflicts or confusion, we can keep this PR open.

**Comment by narve** [comment]
So, the reason a would like to keep block order and curtailable block order out of this PR is that I find the definitions incomplete. The definition on Nordpool is also incomplete and/or inconsistent. The terms are probably based on historical market mechanisms, and then partially updated  to reflect new market scenarios. 

The definitions are mixing several different properties of orders, e.g. price rules and period rules and quantity rules, in an incosistent and incomplete way. 

Furthermore, I have reason to believe that those order types/definitions are not applicable or even not understandable outside of Nordpool/EPEX. This is supposed to be a "Universal" standard after all :) 

I believe we can do better. I will sketch out a proposal here: 

Orders can be categorized along several axes, some of which are: 

- Quantity: Some orders are can be matched-in-part, and some orders need to be fully "filled". This is deduced from the minimumAcceptanceCriteria. Possible categorizations are "Curtailable" (minimumAcceptanceQuantity < Quantity) and "AllOrNothing" (minimumAcceptanceQuantity == Quantity)
- Price: Some orders have a fixed price regardless of quantity, while some orders have price depending on quantity bought (using the piecewise linear price curve concept). Possible categorizations are "FixedPrice" or "VariablePrice". 
- Period: Some orders needs to be matched for the whole period stated in the order (e.g. 1300-1800), while other orders can be divided into time-slots and some of them met (e.g. buying 1300-1500 of an order that has period 1300-1800). Possible categorizations are "SingleSlot" (must match the whole period) or "MultiSlot" (one can match/buy parts of the period). 
- FillType: Depending on the fill type, some orders are killed immediately after matching, while some can remain in the market. Possible categorizations are "Normal" (remains in market), "FillOrKill", "FillAndKill". 

With these "universal" categorizations in place, one can then better communicate how an order behaves. 

For instance, the Nordpool "regular block order" is (as far as I can understand) an order that is Quantity:AllOrNothing, Period:SingleSlot and FillType: FillOrKill. 

What do you think?

**Comment by ArnaudDebray** [comment]
I agree with the approach. I think it makes things clear. Thank you for proposing this. Do you include it in the file or do you want me to do the modification?

The last words that I would like to clarify are "Limit price" and "market price". Shoul it be included in the same way as 5th axe to define an order ? I don't think N-SIDE is really using this in our concept

**Comment by narve** [comment]
> The last words that I would like to clarify are "Limit price" and "market price". Shoul it be included in the same way as 5th axe to define an order ? I don't think N-SIDE is really using this in our concept

Hm, I guess we should say something about Limit and Market, yes.

**Comment by narve** [comment]
Feel free to add this to the PR... I don't think I will have time before todays meeting.

File: glossary.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+**Limit price:** For buy orders, this is the upper limit of what the buyer is willing to pay. 
+For a sell order, this is the lower limit of what the seller is willing to accept. See "Market price". 
+
+**Market price:** For buy orders, this means that the buyer will is willing to pay the current market sell price. For sell orders, this means that the seller is willing to sell at the current market buy price. 
+
+**MeterReading:** Registration of the actual power consumed/injected by an asset
+
+**Order:** Flexibility order placed on a market. The order can be either “Buy” in case the market participant wants to procure flexibility or “Sell” in case the market participant wants to offer flexibility. 
+
+**PayAsBid:** This is a market rule stating that the price at which trade between two offers will occur is the proposed price of the first offer placed on the platform.
```
Comment (cdmNSIDE) [inline glossary.md:34]: Mix between bid / offer / order

**Comment by narve** [comment]
@narve add the essentials of the long comment into the glossary. Add "PricingScheme" as a category and state the legal values (payment types)

**Comment by narve** [comment]
I feel that this PR is ready for merging now. There will probably be updates/modifications later on.

**Review by narve — APPROVED** [review]

File: glossary.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+**Limit price:** For buy orders, this is the upper limit of what the buyer is willing to pay. 
+For a sell order, this is the lower limit of what the seller is willing to accept. See "Market price". 
+
+**Market price:** For buy orders, this means that the buyer will is willing to pay the current market sell price. For sell orders, this means that the seller is willing to sell at the current market buy price. 
+
+**MeterReading:** Registration of the actual power consumed/produced by an asset
+
+**Order:** Flexibility order placed on a market. The order can be either “Buy” in case the market participant wants to procure flexibility or “Sell” in case the market participant wants to offer flexibility. 
+
+**PayAsBid:** This is a market rule stating that the price at which trade between two offers will occur is the proposed price of the first order placed on the platform.
```
Comment (cdmNSIDE) [inline glossary.md:38]: @ArnaudDebray replace "offers" by "orders", no ?

Comment (narve) [reply glossary.md]: fixed

**Comment by cdmNSIDE** [comment]
Thanks @narve. I let Arnaud do a last check but it's ok for me

File: glossary.md [hunk 3]
```diff
@@ -1,21 +1,25 @@
-# Glossary
+# Glossary and abbreviations
 
 **Activation market:** Flexibility market where the assets are dispatched for the 
  period covered by a trade.  
 
-**Baseline:** The forecasted aggregated consumption/injection of an asset portfolio (see portfolio) during a defined time period
+**Asset:** In this context, an asset represents a physical or vertical device which
```
Comment (ArnaudDebray) [inline glossary.md:6]: *virtual

File: glossary.md [hunk 4]
```diff
@@ -27,14 +31,49 @@ For a sell order, this is the lower limit of what the seller is willing to accep
...
 
-**Reservation Market:** Flexibility market in which participants offer the reservation of their flexibility (possibly weeks or years in advance depending on market rules) before actual dispatching. If their reservation offer is activated, they are forced to place an offer on the consecutive activation market (see above)
+**QuantityType:** What is traded on a market / in an order. Possible values are 
+*ActivePower*, *ReactivePower*, *Energy*, *Capacity*. 
+
+**Reservation Market:** Flexibility market in which participants offer the reservation of their flexibility (possibly weeks or years in advance depending on market rules) before actual dispatching.
+The actual dispatching can be done outside the market or in a consecutive activation market. 
+
+**Trade:** A trade either between two participants of a market, or between one participant 
+and the FMO. A trade is always the result of two or more orders being matched.
```
Comment (ArnaudDebray) [inline glossary.md:51]: is it possible to have a trade between a market participant and the FMO ?

File: glossary.md [hunk 5]
```diff
@@ -1,21 +1,25 @@
...
-
-**DSO:** Distribution System Operator
+**DSO:** Distribution System Operator, the entity responsible for operating a electrical grid. 
 
 **Fill-and-kill:** Type of order that is immediately matched. The order is killed and instantly removed after matching, whether it is matched fully, partially or not at all. Available in NODES continuous market. Commonly abbreviated as FaK. Also known as Immediate-or-Cancel (IoC).
 
 **Fill-or-kill:** Type of order that is immediately matched. The order is either matched in full and then killed, or else it is killed and instantly removed. No partial matching is allowed. Available in NODES continuous market. Commonly abbreviated as FoK. 
 
-**FSP:** Flexibility Service Provider  
+**FMO:** Flexible Market Operator, a market where participants can trade flexibility.
```
Comment (ArnaudDebray) [inline glossary.md:20]: I would say : "Flexibility Market Operator, the operator of the platform hosting a market on which participants can trade flexibility"

File: glossary.md (hunk 3 above)
Comment (narve) [reply glossary.md]: fixed

File: glossary.md (hunk 4 above)
Comment (narve) [reply glossary.md:51]: yes, and in NODES that is the default. Each trade is using the FMO as the middle man

File: glossary.md (hunk 5 above)
Comment (narve) [reply glossary.md]: agreed

**Comment by narve** [comment]
ready for merging now?

File: glossary.md (hunk 4 above)
Comment (ArnaudDebray) [reply glossary.md:51]: ok then :)

**Review by ArnaudDebray — COMMENTED** [review]
All good for me :)

**Pull request merged**